recipes/clhep/all/conanfile.py

```diff
@@ -0,0 +1,166 @@
+import glob
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class ClhepConan(ConanFile):
+    name = "clhep"
+    description = "Class Library for High Energy Physics."
+    license = "LGPL-3.0-only"
+    topics = ("conan", "clhep", "cern", "hep", "high energy", "physics", "geometry", "algebra")
+    homepage = "http://proj-clhep.web.cern.ch/proj-clhep"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    short_paths = True
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+        if self.settings.compiler == "Visual Studio" and self.options.shared:
+            raise ConanInvalidConfiguration("CLHEP doesn't properly build its shared libs with Visual Studio")
+        if self.settings.os == "Windows" and self.settings.compiler == "gcc":
+            raise ConanInvalidConfiguration("CLHEP doesn't support MinGW")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.version, self._source_subfolder)
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def package(self):
+        self.copy(pattern="COPYING*", dst="licenses", src=os.path.join(self._source_subfolder, "CLHEP"))
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "bin"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "CLHEP-{}".format(self.version)))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        patterns_to_delete = ["*CLHEP.*", "*CLHEP-{}.*".format(self.version)] # combined lib (duplicate of components libs)
+        patterns_to_delete.extend(["*.a"] if self.options.shared else ["*.so", "*.dylib"])
```

Comment:
Does this work for Visual Studio? And shared Windows builds? (mingw's `.dll.a` and visual studio's `.dll.lib`?)

Reply:
Yes. clhep doesn't export its symbols for shared lib (this is why it's broken), so there is no import lib to delete (bin dir is deleted because dll are created).
clhep doesn't support MinGW.

Reply:
Oops, didn't see the `configure`.